Formula/ooniprobe.rb

```diff
@@ -3,287 +3,163 @@ class Ooniprobe < Formula
-  end
-
-  resource "pypcap" do
-    url "https://files.pythonhosted.org/packages/33/21/d1f24d8a93e4e11bf604d77e04080c05ecb0308a5606936a051bd2b2b5da/pypcap-1.2.2.tar.gz"
-    sha256 "a32322f45d63ff6196e33004c568b9f5019202a40aa2b16008b7f94e7e119c1f"
-
-    # https://github.com/pynetwork/pypcap/pull/79
-    # Adds support for the new CLT SDK with the 10.x
-    # series of development tools.
-    patch do
-      url "https://github.com/pynetwork/pypcap/commit/7c2a570823eeb45b0daf69960867d498faa6dd87.patch?full_index=1"
-      sha256 "cb0c9b271d293e49e504793bed296e0fa73cca546dbc2814e0ea01351e66d9b2"
-    end
-  end
-
-  resource "PyYAML" do
-    url "https://files.pythonhosted.org/packages/4a/85/db5a2df477072b2902b0eb892feb37d88ac635d36245a72a6a69b23b383a/PyYAML-3.12.tar.gz"
-    sha256 "592766c6303207a20efc445587778322d7f73b161bd994f227adaa341ba212ab"
-  end
-
-  resource "scapy" do
-    url "https://files.pythonhosted.org/packages/ac/14/c792a14b9f8bc4bb9c74c0594c167a2da36e31964098d9e27202142cbd7d/scapy-2.3.3.tgz"
-    sha256 "9d3b0293dcdc2cc42eedacbc9003038109558c4b5c5e4a3fa01b8ef5762f1eb0"
-  end
-
-  resource "service_identity" do
-    url "https://files.pythonhosted.org/packages/f3/2a/7c04e7ab74f9f2be026745a9ffa81fd9d56139fa6f5f4b4c8a8c07b2bfba/service_identity-16.0.0.tar.gz"
-    sha256 "0630e222f59f91f3db498be46b1d879ff220955d7bbad719a5cb9ad14e3c3036"
-  end
-
-  resource "six" do
-    url "https://files.pythonhosted.org/packages/b3/b2/238e2590826bfdd113244a40d9d3eb26918bd798fc187e2360a8367068db/six-1.10.0.tar.gz"
-    sha256 "105f8d68616f8248e24bf0e9372ef04d3cc10104f1980f54d57b2ce73a5ad56a"
-  end
-
-  resource "Twisted" do
-    url "https://files.pythonhosted.org/packages/d2/5d/ed5071740be94da625535f4333793d6fd238f9012f0fee189d0c5d00bd74/Twisted-17.1.0.tar.bz2"
-    sha256 "dbf211d70afe5b4442e3933ff01859533eba9f13d8b3e2e1b97dc2125e2d44dc"
-  end
-
-  resource "txsocksx" do
-    url "https://files.pythonhosted.org/packages/ed/36/5bc796eb2612b500baa26a68481d699e08af5382152a9de18e5a45b44ea7/txsocksx-1.15.0.2.tar.gz"
-    sha256 "4f79b5225ce29709bfcee45e6f726e65b70fd6f1399d1898e54303dbd6f8065f"
-  end
-
-  resource "txtorcon" do
-    url "https://files.pythonhosted.org/packages/03/23/4453ab8728c84963cf293d5180b9f529bb10bf8285031c4681e2621b175f/txtorcon-0.18.0.tar.gz"
-    sha256 "12be80f1d5e2893378c6e8c752cf159479f868f8424e16b34b75cd679a0ab171"
-  end
-
-  resource "Werkzeug" do
-    url "https://files.pythonhosted.org/packages/13/a2/c4f2a1e1e9239cd979de00a2d7e0008559c46d920e9842e9b8063c5e6bf5/Werkzeug-0.12.tar.gz"
-    sha256 "f007848ed997101cb5c09a47e46c0b0b6f193d0f8a01cd2af920d77bf1ab4e68"
-  end
-
-  resource "zope.interface" do
-    url "https://files.pythonhosted.org/packages/44/af/cea1e18bc0d3be0e0824762d3236f0e61088eeed75287e7b854d65ec9916/zope.interface-4.3.3.tar.gz"
-    sha256 "8780ef68ca8c3fe1abb30c058a59015129d6e04a6b02c2e56b9c7de6078dfa88"
-  end
+  depends_on "go@1.14" => :build
```

Comment:
Does it not compile with the latest Go version?

Reply:
It does build also with the latest version of golang, but we do extensive testing with this specific version.

@bassosimone can maybe add more details about this aspect.

Reply:
It does compile with Go 1.15. Yet, with Go 1.15, one of our dependencies (the psiphon censorship circumvention system; https://github.com/Psiphon-Labs/psiphon-tunnel-core) does not support using QUIC for censorship evasion.

Psiphon will migrate to Go 1.15 when Go 1.14 is going EOL. Currently Psiphon builds its own apps and generally always uses Go 1.14 for its circumvention system.

For this reason, we're using Go 1.14 such that our testing of Psiphon is closer to how Psiphon is usually used (and specifically such that it can use QUIC as an evsasion mechanism).

Reply:
Would it be okay for us adding a comment explaining why we're using Go 1.14?

Reply:
If it works with go 1.15 it should use go 1.15, such are the implications of a evergreen package-manager like Homebrew

Reply:
@SMillerDev using go 1.15 leads to our upstream dependencies breaking and some missed functionality. 

As soon our upstream fixes this, we will be sure to further revise the formula to run with golang 1.14.

Is it OK in this case to keep this as-is?

Reply:
Since it's essentially a new formula I'd like for the formula to use the latest go version before merging. This way we avoid adding the formula with some bit rot build in.